test/dns.server.test.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  createHostedZone,
+  createRecord,
+  upsertRecord,
+  deleteRecord,
+  getChangeStatus,
+  RecordType,
+} from "../app/lib/dns.server";
+
+describe("DNS server lib function test", () => {
+  let hostedZoneId: string | undefined;
```

Comment:
Let's use `string` vs. `string | undefined` which should allow you to get rid of the `!` on all your uses of `hostedZoneId` below.